official/resnet/resnet_run_loop.py

```diff
@@ -245,14 +252,22 @@ def resnet_model_fn(features, labels, mode, model_class,
     EstimatorSpec parameterized according to the input params and the
     current mode.
   """
-
   # Generate a summary node for the images
   tf.summary.image('images', features, max_outputs=6)
-  # TODO(tobyboyd): Add cast as part of input pipeline on cpu and remove.
-  features = tf.cast(features, dtype)
-
-  model = model_class(resnet_size, data_format, resnet_version=resnet_version,
-                      dtype=dtype)
+  # Checks that features/images have same data type being used for calculations.
+  assert features.dtype == dtype
+
+  if use_keras_model:
+    model = keras_resnet_model.ResNet50(classes=num_classes, weights=None,
+                                        training=(mode == tf.estimator.ModeKeys.TRAIN))
+    bn_updates = []
+    if mode == tf.estimator.ModeKeys.TRAIN:
+      for l in model.layers:
+        if 'bn' in l.name:
```

Comment:
Collecting updates by matching patters on the name seems pretty fragile.
Can't this be:

```python
if mode == tf.estimators.ModeKeys.TRAIN:
  bn_updates = model.get_updates_for(features)
```

or perhaps (since `Model` doesn't seem like it recurses into its layers for `get_updates_for`):

```python
if mode == tf.estimators.ModeKeys.TRAIN:
  bn_updates.append(l.get_updates_for(features) for l in model.layers)
```
?